core/icons/comment_icon.ts

```diff
@@ -218,7 +221,18 @@ export class CommentIcon extends Icon implements IHasBubble, ISerializable {
    */
   onTextChange(): void {
     if (this.textInputBubble) {
-      this.text = this.textInputBubble.getText();
+      const newText = this.textInputBubble.getText();
```

Comment:
Just realized we should also update `setText` to fire an event. Could you do that as well?

Reply:
Yea, I can get both of these things resolved!

Reply:
Alright @BeksOmega, should be good to go!

Reply:
Oh hm my edit to that comment didn't save, sorry about that. `setText` fixes look great! But we also need to make a little change to [`setCommentText`](https://github.com/google/blockly/blob/12ac35829e11417e5acda818949fa35a15e9ac74/core/block.ts#L2224) to only fire the event if the icon is removed. Otherwise we're going to get duplicate events!

Once that's in, I'll give this a manual test, and hopefully should be good to go =)

Thank you for your work on this @treypisano !

Reply:
Ok I think I got it. Thanks for helping me out with this.